### PR TITLE
feat: Dropdown and TextFeild accessibility improvements

### PR DESCRIPTION
**Pull request by or-nuri-monday**
#### Basic

- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks.
- [ ] Component is written in TypeScript.

#### Style

- [ ] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
- [ ] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-style) tokens
- [ ] Displays correctly in all the themes

#### Storybook

- [ ] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.mdx` file.
- [ ] Stories include all flows of using the component.
- [ ] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

#### Accessibility

- [ ] Component is accessible.
- [ ] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--docs)

### CODE REVIEW

File: src/components/TextField/TextFieldConstants.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ export enum TextFieldTextType {
   SEARCH = "search",
   DATE = "date",
   DATE_TIME = "datetime-local",
-  NUMBER = "number"
+  NUMBER = "number",
+  EMAIL = "email"
```
Comment (talkor) [inline src/components/TextField/TextFieldConstants.ts:32]: Please lets separate it to a different PR 🙏🏼

File: src/components/Dropdown/components/menu/menu.jsx [hunk 2]
```diff
@@ -17,20 +17,22 @@ const Menu = ({ children, Renderer, selectProps, dropdownMenuWrapperClassName, .
...
-          [styles.dropdownMenuWrapperFixedPosition]: withFixedPosition,
-          [styles.withoutAnimation]: withoutAnimation
-        },
-        dropdownMenuWrapperClassName
-      )}
-    >
-      {Renderer && Renderer(rendererProps)}
-      {!Renderer && children}
-    </components.Menu>
+    <div id={id} role="listbox" aria-label={ariaLabel}>
```
Comment (talkor) [inline src/components/Dropdown/components/menu/menu.jsx:20]: @shaharzil managed to upgrade to latest react-select version which already supports it out-of-the-box, let's sync about it, it might save us these changes

File: src/components/Dropdown/Dropdown.jsx [hunk 3]
```diff
@@ -332,6 +345,10 @@ const Dropdown = forwardRef(
         aria-readonly={readOnly}
         aria-label={overrideAriaLabel}
         aria-details={tooltipContent}
+        aria-expanded={!readOnly && menuIsOpen}
+        aria-haspopup="listbox"
+        aria-activedescendant
+        role={"combobox"}
```
Comment (talkor) [inline src/components/Dropdown/Dropdown.jsx:351]:
```suggestion
        role="combobox"
```

File: src/components/Dropdown/Dropdown.jsx [hunk 4]
```diff
@@ -404,6 +421,8 @@ Dropdown.defaultProps = {
   tabIndex: "0",
   onOptionRemove: undefined,
   id: DROPDOWN_ID,
+  menuId: DROPDOWN_MENU_ID,
```
Comment (talkor) [inline src/components/Dropdown/Dropdown.jsx:424]: See TS error, also for `menuAriaLabel`

File: src/components/Dropdown/components/option/option.jsx [hunk 5]
```diff
@@ -10,15 +10,17 @@ const Option = ({ Renderer, data, children, optionWrapperClassName, ...props })
...
-      {Renderer ? (
-        <components.Option {...rendererProps} className={cx(styles.optionReset, optionWrapperClassName)}>
-          <Renderer {...rendererProps} {...data} /> {/* Spreading data here for a backward compatability */}
-        </components.Option>
-      ) : (
-        <components.Option {...rendererProps} className={cx(styles.optionReset, optionWrapperClassName)}>
-          <ChildrenContent data={data}>{children}</ChildrenContent>
-        </components.Option>
-      )}
+      <div role="option" aria-selected={props.isSelected}>
```
Comment (talkor) [inline src/components/Dropdown/components/option/option.jsx:13]: Same comment about react-select, this is a breaking change

File: src/components/Dropdown/components/menu/menu.jsx (hunk 2 above)
Comment (shaharzil) [reply src/components/Dropdown/components/menu/menu.jsx:20]: Hi @or-nuri-monday, I played with the react-select library and found this prop (called innerProp) to add props to the menu wrapper:
image

Can you please try it and let us know if it solves the issue? 🙏

File: src/components/Dropdown/components/option/option.jsx (hunk 5 above)
Comment (shaharzil) [reply src/components/Dropdown/components/option/option.jsx:13]: Same comment, this is how to use it in the option component: 
image

File: src/components/Dropdown/Dropdown.jsx [hunk 6]
```diff
@@ -191,7 +201,10 @@ const Dropdown = forwardRef(
       [finalOptionRenderer, optionWrapperClassName]
     );
 
-    const Input = useCallback(props => <components.Input {...props} aria-label="Dropdown input" />, []);
+    const Input = useCallback(
+      props => <components.Input {...props} aria-label="Dropdown input" aria-controls={menuId} />,
+      []
```
Comment (talkor) [inline src/components/Dropdown/Dropdown.jsx:206]: Notice the lint error

File: src/components/TextField/TextFieldConstants.ts (hunk 1 above)
Comment (talkor) [reply src/components/TextField/TextFieldConstants.ts:32]: Lets remove it for now

File: src/components/TextField/TextField.tsx [hunk 7]
```diff
@@ -124,6 +126,7 @@ const TextField: VibeComponent<TextFieldProps, unknown> & {
       trim = false,
       role = "",
       required = false,
+      requiredErrorText = DEFAULT_REQUIRED_ERROR_TEXT,
```
Comment (talkor) [inline src/components/TextField/TextField.tsx:129]: I think it should be customizable using the validations object

File: src/components/TextField/TextField.tsx [hunk 8]
```diff
@@ -135,18 +138,33 @@ const TextField: VibeComponent<TextFieldProps, unknown> & {
...
       [dataTestId, backwardCompatibilityDataTestId],
       getTestId(ComponentDefaultTestId.TEXT_FIELD, id)
     );
     const inputRef = useRef(null);
     const mergedRef = useMergeRef(ref, inputRef, setRef);
 
+    const onBlurCallback = useCallback(
+      (e: React.FocusEvent<HTMLInputElement>) => {
+        if (required && !e.target.value) {
+          setRequiredAndEmpty(true);
```
Comment (talkor) [inline src/components/TextField/TextField.tsx:153]: What about enter? Shouldn't it also be called?

File: src/components/TextField/TextField.tsx (hunk 7 above)
Comment (or-nuri-monday) [reply src/components/TextField/TextField.tsx:129]: But as a user, you can't know if the text field is empty and required. This is also a specific use case. The logic is held inside the component itself

File: src/components/TextField/TextFieldConstants.ts (hunk 1 above)
Comment (or-nuri-monday) [reply src/components/TextField/TextFieldConstants.ts:32]: We need this for another question. I can create another PR but we need this

**Review by talkor — APPROVED** [review]
